aten/src/ATen/native/ComplexHelper.h

```diff
@@ -18,23 +18,15 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
 inline Tensor view_complex_as_float(const Tensor& self) {
   TORCH_INTERNAL_ASSERT(self.is_complex());
   auto new_sizes = self.sizes().vec();
+  const auto float_type = c10::toValueType(self.scalar_type());
+
   // last dimension will always have two elements containing the real and imag vals
   new_sizes.emplace_back(2);
   if (self.numel() == 0) {
-    if(self.scalar_type() == at::kComplexFloat) {
-      return at::empty(new_sizes, self.options().dtype(at::kFloat));
-    } else {
-      return at::empty(new_sizes, self.options().dtype(at::kDouble));
-    }
+    return at::empty({0}, self.options().dtype(float_type));
   } else {
     auto new_strides = computeStrideForComplex(self.strides());
-    if(self.scalar_type() == at::kComplexFloat) {
-      float* data = reinterpret_cast<float*>(self.data_ptr<std::complex<float>>());
-      return at::from_blob(data, new_sizes, new_strides, self.options().dtype(at::kFloat));
-    } else {
-      double* data = reinterpret_cast<double*>(self.data_ptr<std::complex<double>>());
-      return at::from_blob(data, new_sizes, new_strides, self.options().dtype(at::kDouble));
-    }
+    return at::empty({0}, self.options().dtype(float_type)).set_(self.storage(), 0, new_sizes, new_strides);
```

Comment:
I think you want to use the storage offset from self here and not `0`. The offset is already included when you take data_ptr but is not when you take the storage.

Reply:
good catch! updated the input storage offset and added a test as well